app/actions/views/channel.js

```diff
@@ -409,7 +415,13 @@ export function handleSelectChannelByName(channelName, teamName) {
         const response = await dispatch(getChannelByNameAndTeamName(teamName || currentTeamName, channelName));
         const {error, data: channel} = response;
         const currentChannelId = getCurrentChannelId(state);
+        const reachable = channelIsReachable(state, channelName, teamName);
+
+        if (!reachable && errorHandler) {
+            errorHandler();
```

Comment:
```suggestion
            errorHandler(error);
            return {error};
```

Reply:
Wouldn't this will surface the generic API  error response over the desired messaging supplied in the error handling callback?